### ISSUE TITLE
Quality Control Issue Bonanza

### ISSUE THREAD

**Issue by crazyhottommy** [datacarpentry/wrangling-genomics#23]
#### Quality Control

[link](https://github.com/datacarpentry/wrangling-genomics/blob/gh-pages/_episodes/00-readQC.md)  

- [x] Fix:
Teaching: 0 min
Exercises: 0 min

Are my data good enough?  

Objectives:   
•	Clean (trim adaptors and low quality bases of the) FastQ reads using Trimmommatic  
•	Employ for loops to clean multiple FastQ files.

>The first step in the variant calling work flow is to take the FASTQ files received from the sequencing facility and assess the quality of the sequence reads.

- [x] Workflow or work flow?

>Although it looks complicated (and maybe it is), its easy to understand the fastq format with a little decoding. Some rules about the format include…

- [x] It is 

>so for example in our data set, one complete read is:

For example,

```bash
$ head -n4 SRR098281.fastq 
```

- [x] Need to download the data first before the class?
- [x] A detailed instuction needed.

```
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS.....................................................
..........................XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX......................
...............................IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII......................
.................................JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ......................   LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL....................................................   !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~   |                         |    |        |                              |                     |  33                        59   64       73                            104                   126   0........................26...31.......40                                
                       -5....0........9.............................40 
                             0........9.............................40 
                                3.....9.............................40    0.2......................26...31........41  
```

- [x] Need to color the scores as in the wiki page. I do not know how to add colors..maybe a direct screen shot works? A link to the fastq wiki page is needed.

https://en.wikipedia.org/wiki/FASTQ_format

>Using the quality encoding character legend, the first nucleotide in the read (C) is called with a quality score of 31 and our Ns are called with a score of 2. As you can tell by now, this is a bad read.

- [x] How to define a bad read? quality low and many Ns?

- [x] need to explain phred + 33, phred + 64

### Running FASTQC 
- [x] Some text here (need to be filled)

>The sample data we will be working with is in a hidden directory (placing a ‘.’ in front of a directory name hides the directory. In the next step we will move some of those hidden files into our new dirctories to start our project.

- [x] why the data need to be hidden?

### B. Run FastQC

- [x] need to download FASTQC first. (saw it http://www.datacarpentry.org/wrangling-genomics/setup/)

- [x] should we provide the output directory in the fastqc commands (`-o ~/dc_workshop/results/fastqc_untrimmed_reads/`) rather than generate the results in the current `data` folder and then move the results to the results folder?

### How to clean reads using Trimmomatic
- [x] Some text here (need to be filled)

#### A. detailed explanation of features

- [x] Need to provide a `ILLUMINACLIP:SRR_adapters.fa` file in the repo

>java -jar calls the Java program, which is needed to run trimmomargumentstic

- [x] trimmomargumentstic?  

```bash
$ for infile in *.fastq
>do
>outfile=$infile\_trim.fastq
>java -jar ~/Trimmomatic-0.32/trimmomatic-0.32.jar SE $infile $outfile SLIDINGWINDOW:4:20 MINLEN:20
>done
```

- [ ] always double quote your bash variable and use {}

```bash
$ for infile in *.fastq
>do
>outfile="${infile}"_trim.fastq
>java -jar ~/Trimmomatic-0.32/trimmomatic-0.32.jar SE $infile $outfile SLIDINGWINDOW:4:20 MINLEN:20
>done
```
### Variant Calling Workflow

Objectives 
 
 
* Use a For (for) loop from the previous lesson to help automate repetitive tasks  

* Becoming (Become) familiar with data formats encountered during variant calling

Your directory structure should now look like this:

```
dc_workshop
├── data
    ├── ref_genome
        └── ecoli_rel606.fasta
    ├── untrimmed_fastq
    └── trimmed_fastq
        ├── SRR097977.fastq
        ├── SRR098026.fastq
        ├── SRR098027.fastq
        ├── SRR098028.fastq
        ├── SRR098281.fastq
        └── SRR098283.fastq
 ├── results
 └── docs

```

- [x] A `tree` command need to be introduced?

>We perform read alignment or mapping to determine where in the genome our reads originated from. There are a number of tools to choose from and while there is no gold standard there are some tools that are better suited for particular NGS analyses.

- [x] >We perform read alignment or mapping to determine where in the genome our reads originated from. There are a number of tools to choose from and while there is no gold standard, there are some tools that are better suited for particular NGS analyses.

### Index the reference genome

```bash
$ bwa index data/ref_genome/ecoli_rel606.fasta     # This step helps with the speed of alignment
```

- [x] How long it takes for a ecoli genome? my experience with human genome is about 2 hours.

Explore the information within your SAM file:

```bash
$ head results/sam/SRR097977.aligned.sam
```

- [x] `less -S` is better?

```bash
$ less -S results/sam/SRR097977.aligned.sam
```

**Comment by lexnederbragt** [comment]
> need to download FASTQC first. (saw it http://www.datacarpentry.org/wrangling-genomics/setup/)

Can't we assume that at this point all software is installed  and running?

> A `tree` command need to be introduced?

Not all systems have the `tree` command, but it may be mentioned.

> How long it takes for a ecoli genome? my experience with human genome is about 2 hours.

It takes very little time.

> `less -S` is better?

Yes, provided we explain it.

**Comment by crazyhottommy** [comment]
yes, I think all the tools are pre-installed according to a later lesson:
http://www.datacarpentry.org/wrangling-genomics/01-variant_calling_workflow/

>NOTE: All of the tools that we will be using in this workflow have been pre-installed on our remote computer

**Comment by JasonJWilliamsNY** [comment]
All tools and data should be on the image. All of these lessons were intended to be taught from a cloud instance.

**Comment by fpsom** [comment]
@JasonJWilliamsNY that's a good point! I don't think that is clear - and in any case, cloud is not mentioned in the Genomics Workshop with Pipeline Workflow (http://www.datacarpentry.org/genomics-workshop/).

**Comment by JasonJWilliamsNY** [comment]
@crazyhottommy the .hidden data are data in the image, namely the actual fastq file we do the calling on. They are hidden to reduce the chance that a learner accidentally manipulates them somewhere earlier on in the workshop.

**Comment by JasonJWilliamsNY** [comment]
@crazyhottommy 

> should we provide the output directory in the fastqc commands (-o ~/dc_workshop/results/fastqc_untrimmed_reads/) rather than generate the results in the current data folder and then move the results to the results folder?

I think it was a good way to reinforce their shell commands. Maintainers can make that call.

**Comment by JasonJWilliamsNY** [comment]
@crazyhottommy 

> Need to provide a ILLUMINACLIP:SRR_adapters.fa file in the repo

Files we are working with don't have adapters, so that's why we skipped this.

**Comment by ErinBecker** [comment]
Thanks for putting this together @crazyhottommy. I've added checkboxes to the original issue and can go through and put in PR(s) for any of the checkboxes not already resolved in the comments above. Will be working on that today.

**Comment by ErinBecker** [comment]
Moved one from list to new issue: https://github.com/datacarpentry/wrangling-genomics/issues/23

Closing others.

**Comment by ErinBecker** [comment]
@crazyhottommy - I think I've addressed all of these except the one I moved to #23. Please let me know if I've missed anything!

**Comment by crazyhottommy** [comment]
Thanks! closing this then.
